app/helpers/api_page_helper.rb

```diff
@@ -0,0 +1,50 @@
+module ApiPageHelper
+  # ask for current_weather
+  def get_weather(city = 'berlin', api_id)
+    # Rails.cache.clear
+    if api_id == 1
+      cache_name = 'curr_weather'
+    elsif api_id == 2
+      cache_name = 'forecast_weather'
+    end
+
+    exists_weather_of(city, api_id, cache_name)
+    if api_id == 1
+      get_curr_weather_data(Rails.cache.fetch('curr_weather'))
```

Comment:
Yes, i thought about this and i fixed this in exists_weather_of(there i check if the city anem is the same!). But tour is a nice approach for a later version, so i can cache different cities and not just one at a time!